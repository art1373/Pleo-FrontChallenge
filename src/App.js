import React, { useState, useEffect } from "react";
import "./App.css";

import { ExpenseContext } from "./Expenses/context/ExpensesContext";
import UserExpenses from "./Expenses/containers/userExpenses";
import Axios from "axios";
import Routes from "./api/Routes";

function App() {
  const [expenses, setExpenses] = useState([]);
  useEffect(() => {
    Axios.get(`${Routes.GetExpenses}`)
      .then(res => {
        let json = res.data.expenses;
        setTimeout(() => {
          return setExpenses(json);
        }, 2000);
      })
      .catch(err => {
        console.log(err.response);
      });
  }, []);

  return (
    <div className="App">
      <ExpenseContext.Provider value={{ expenses }}>
        <UserExpenses />
      </ExpenseContext.Provider>
    </div>
  );
}

export default App;
