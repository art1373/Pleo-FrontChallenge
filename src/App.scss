$color-white: #f3f4f6;
$color-pleo: #f36;
$color-blue: #2c86ff;

.App {
  text-align: center;
}

body {
  background: #f8f9fb;
}

@font-face {
  font-family: "Berlin Sans";
  src: url("./assets/fonts/BerlinSans.ttf");
}
@font-face {
  font-family: "TwCent";
  src: url("./assets/fonts/TwCent.otf");
}
@font-face {
  font-family: "Dubai";
  src: url("./assets/fonts/DubaiRegular.ttf");
}

#page-wrap {
  text-align: center;

  /* Prevent sidebar from showing a scrollbar on page */
  overflow: auto;
}

/* Individual item */
.bm-item {
  display: inline-block;

  /* Our sidebar item styling */
  text-decoration: none;
  margin-bottom: 10px;
  color: $color-white;
  transition: color 0.2s;
}

/* Change color on hover */
.bm-item:hover {
  color: $color-blue;
}

/* The rest copied directly from react-burger-menu docs */

/* Position and sizing of burger button */
.bm-burger-button {
  position: fixed;
  width: 36px;
  height: 30px;
  left: 36px;
  top: 6px;
}

/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: $color-pleo;
}

/* Position and sizing of clickable cross button */
.bm-cross-button {
  height: 24px;
  width: 24px;
}

/* Color/shape of close button cross */
.bm-cross {
  background: black;
}

/* General sidebar styles */
.bm-menu {
  background: $color-pleo;
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
}

/* Morph shape necessary with bubble or elastic */
.bm-morph-shape {
  fill: $color-pleo;
}

/* Wrapper for item list */
.bm-item-list {
  color: #f8f9fb;
}

/* Styling of overlay */
.bm-overlay {
  background: rgba(0, 0, 0, 0.3);
}
