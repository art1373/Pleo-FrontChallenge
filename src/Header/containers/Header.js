import React, { useContext } from "react";
import "../styles/styles.scss";
import { ExpenseContext } from "../../Expenses/context/ExpensesContext";

import { MdRefresh } from "react-icons/md";
const Header = props => {
  const ExpenseProvider = useContext(ExpenseContext);

  return (
    <div className="container">
      <div className="content">
        <p>Total Expenses: {ExpenseProvider.expenses.length}</p>
        <p>
          £
          {ExpenseProvider.expenses.reduce((total, exp) => {
            return exp.amount.currency === "GBP"
              ? Math.round(Number(total) + Number(exp.amount.value))
              : Math.round(Number(total));
          }, 0)}
        </p>
        <p>
          €
          {ExpenseProvider.expenses.reduce((total, exp) => {
            return exp.amount.currency === "EUR"
              ? Math.round(Number(total) + Number(exp.amount.value))
              : Math.round(Number(total));
          }, 0)}
        </p>
        <p>
          Kr.
          {ExpenseProvider.expenses.reduce((total, exp) => {
            return exp.amount.currency === "DKK"
              ? Math.round(Number(total) + Number(exp.amount.value))
              : Math.round(Number(total));
          }, 0)}
        </p>
        <div style={{ marginTop: "10px" }}>
          <MdRefresh
            size={20}
            style={{ cursor: "pointer" }}
            onClick={() => window.location.reload()}
          />
        </div>
      </div>
    </div>
  );
};

export default Header;
