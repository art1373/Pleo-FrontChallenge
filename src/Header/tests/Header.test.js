import React from "react";
import { shallow } from "enzyme";
import Header from "../containers/Header";
import App from "../../App";
import renderer from 'react-test-renderer';


describe("Header component", () => {
  it("shows expenses length in header", () => {
    setTimeout(() => {
      const wrapper = shallow(<Header />);
      const lengt = wrapper.find("p").text();
      expect(lengt.toEqual(`Total Expenses: ${25}`));
    }, 3000);
  });
});

it("matches the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
