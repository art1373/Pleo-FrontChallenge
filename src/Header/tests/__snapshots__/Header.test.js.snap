// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <p>
        Total Expenses: 
        0
      </p>
      <p>
        £
        0
      </p>
      <p>
        €
        0
      </p>
      <p>
        Kr.
        0
      </p>
      <div
        style={
          Object {
            "marginTop": "10px",
          }
        }
      >
        <svg
          fill="currentColor"
          height={20}
          onClick={[Function]}
          size={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "cursor": "pointer",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div>
    <div
      className="bm-overlay"
      onClick={[Function]}
      style={
        Object {
          "MozTransform": "translate3d(100%, 0, 0)",
          "MsTransform": "translate3d(100%, 0, 0)",
          "OTransform": "translate3d(100%, 0, 0)",
          "WebkitTransform": "translate3d(100%, 0, 0)",
          "background": "rgba(0, 0, 0, 0.3)",
          "height": "100%",
          "opacity": 0,
          "position": "fixed",
          "transform": "translate3d(100%, 0, 0)",
          "transition": "opacity 0.3s, transform 0s 0.3s",
          "width": "100%",
          "zIndex": 1000,
        }
      }
    />
    <div
      className="bm-menu-wrap"
      id=""
      style={
        Object {
          "MozTransform": "translate3d(-100%, 0, 0)",
          "MsTransform": "translate3d(-100%, 0, 0)",
          "OTransform": "translate3d(-100%, 0, 0)",
          "WebkitTransform": "translate3d(-100%, 0, 0)",
          "height": "100%",
          "position": "fixed",
          "right": "inherit",
          "transform": "translate3d(-100%, 0, 0)",
          "transition": "all 0.3s",
          "width": "300px",
          "zIndex": 1100,
        }
      }
    >
      <div
        className="bm-morph-shape"
        style={
          Object {
            "MozTransform": "",
            "MsTransform": "",
            "OTransform": "",
            "WebkitTransform": "",
            "height": "100%",
            "left": "inherit",
            "position": "absolute",
            "right": 0,
            "transform": "",
            "width": 120,
          }
        }
      >
        <svg
          height="100%"
          preserveAspectRatio="none"
          viewBox="0 0 100 800"
          width="100%"
        >
          <path
            d="M-1,0h101c0,0-97.833,153.603-97.833,396.167C2.167,627.579,100,800,100,800H-1V0z"
          />
        </svg>
      </div>
      <div
        className="bm-menu"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "100%",
            "overflow": "visible",
            "position": "fixed",
            "right": "inherit",
            "whiteSpace": "nowrap",
            "width": 180,
          }
        }
      >
        <nav
          className="bm-item-list"
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <a
            className="bm-item menu-item"
            href="/"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Home
          </a>
          <a
            className="bm-item menu-item"
            href="/expenses"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Expenses
          </a>
          <a
            className="bm-item menu-item"
            href="/Cards"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Virtual Cards
          </a>
          <a
            className="bm-item menu-item"
            href="/About"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            About Pleo
          </a>
        </nav>
      </div>
      <div
        style={Object {}}
      >
        <div
          className="bm-cross-button"
          style={
            Object {
              "height": 24,
              "position": "absolute",
              "right": 8,
              "top": 8,
              "width": 24,
            }
          }
        >
          <span
            style={
              Object {
                "position": "absolute",
                "right": "14px",
                "top": "6px",
              }
            }
          >
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(45deg)",
                  "width": 3,
                }
              }
            />
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(-45deg)",
                  "width": 3,
                }
              }
            />
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "cursor": "pointer",
                "fontSize": 0,
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Close Menu
          </button>
        </div>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="bm-burger-button"
        style={
          Object {
            "zIndex": 1000,
          }
        }
      >
        <span>
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "0%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "40%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "80%",
              }
            }
          />
        </span>
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "cursor": "pointer",
              "fontSize": 0,
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          Open Menu
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": 150,
        "transition": "all 1s",
      }
    }
  >
    <div
      style={
        Object {
          "background": "rgba(255, 51, 102, 1)",
          "border": "none",
          "borderRadius": 15,
          "color": "rgba(255, 51, 102, 1)",
          "fontFamily": "TwCent",
          "fontSize": "1rem",
          "height": "2rem",
          "margin": "auto",
          "outline": "none",
          "textAlign": "center",
          "transition": "all 1s",
          "width": "50%",
        }
      }
    >
      <input
        id="fltr"
        onChange={[Function]}
      />
      <img
        alt="search"
        className="searchIcon"
        src="search.svg"
      />
    </div>
    <select
      id="Currency"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All Currencies
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="DKK"
      >
        DKK
      </option>
    </select>
    <div
      className="sc-fMiknA gArxby"
      color="#f36"
      size="default"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "margin": "auto",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </div>
</div>
`;
